### DISCUSSION TITLE
Issues with Numpy when using rio-tiler in AWS Lambda

### DISCUSSION THREAD

**Discussion by abdul-imran** [cogeotiff/rio-tiler#739] [Q&A]
- Created a lambda layer using Python 3.12
`pip install rio-tiler pillow -t lambda_layer/python`

- Created a lambda function which reads a Cloud Optimized GeoTiff (COG) from AWS S3 & returns tiles based on the values for x,y, & z. The below statements give me an issue
`
from rio_tiler.io import COGReader

from rio_tiler.models import Tile

from rio_tiler.utils import render
`

- Geting the below error while testing 
`Unable to import module 'lambda_function': Error importing numpy: you should not try to import numpy from\n        its source directory; please exit the numpy source tree, and relaunch\n        your python interpreter from there.`

**Comment by vincentsarago** [comment]
this is not a rio-tiler issue, the error message seems to indicate you have an issue with Numpy or something that happened during packaging your lambda function

**Comment by abdul-imran** [comment]
Hey @vincentsarago  - Thank you. I found the solution for this.
I had to create my Lambda package in an amazon Linux2 instance & also install expat, expat-devel to fix this.

```
pip3.12 install rio-tiler pillow -t lambda_layer/python

sudo yum install expat expat-devel

mkdir -p lambda_layer/lib

cp /usr/lib64/libexpat.so.1* lambda_layer/lib/
```

**Reply by vincentsarago** [reply]
you may also need to install gcc and g++ 

https://github.com/developmentseed/titiler/blob/main/deployment/aws/lambda/Dockerfile#L7-L8
